indico/htdocs/js/indico/modules/contributions/types_dialog.js

```diff
@@ -0,0 +1,61 @@
+(function(global) {
+    'use strict';
+
+    global.setupContributionTypesDialog = function setupContributionTypesDialog() {
+        var $manageContributionTypes = $('.manage-contribution-types');
+
+        /* Set the customData to true indicating that the contribution list needs to be refreshed */
+        function dialogModified() {
+            $manageContributionTypes.trigger('ajaxDialog:setData', [true]);
+        }
+
+        $('.manage-contribution-types table').tablesorter({
+            sortList: [[0, 0]],
+            headers: {
+                2: {
+                    sorter: false
+                }
+            }
+        });
+
+        $('.js-new-contribution-type').on('ajaxDialog:closed', function(evt, data) {
+            evt.preventDefault();
+            if (data) {
+                var $lastRow = $('.manage-contribution-types table tr:last');
+                if ($lastRow.length) {
+                    $lastRow.after(data.html_row);
+                } else {
+                    $manageContributionTypes.trigger('ajaxDialog:reload');
+                }
+            }
+        });
+
+        $manageContributionTypes.on('ajaxDialog:closed', '.js-edit-contribution-type', function(evt, data) {
+            evt.preventDefault();
+            var $row = $(this).closest('tr');
```

Comment:
You are using it only if `data` is not empty. It should be better to move it inside the `if`.

`$(this).closest('tr').replaceWith(data.html_row);`